### DISCUSSION TITLE
Ruby Metrics SDK Stable ETA

### DISCUSSION THREAD

**Discussion by GBoucher42** [open-telemetry/opentelemetry-ruby#1743] [General]
Hi, I see there is a work being done on the metrics SDK based on the last few releases, is there a rough ETA as to when it would be stable and up to par with other languages?

Thank you

**Comment by kaylareopelle** [comment]
Hi @GBoucher42, thanks for reaching out! This is a great question.

I think this is something the SIG should discuss during our next meeting. If you'd like to join, we meet Tuesdays at 9:00 AM Pacific time. You can get the invite link from the [opentelemetry-community repo](https://github.com/open-telemetry/community?tab=readme-ov-file#implementation-sigs).

I'll add more to this conversation after the meeting. 

In the meantime, if you're curious about the status of various metrics features, I have a first attempt at conveying spec compliance in this [draft PR with a markdown table](https://github.com/open-telemetry/opentelemetry-ruby/pull/1746/files?short_path=42237ae#diff-42237aee3b75ea95ec0ec6357efa1a7cb68ba82ab4386e21a8ec21c75aa57d55).

**Reply by jensljungblad-kisi** [reply]
I see this is a couple years old but I have the same question, we are eyeing a shift to OpenTelemetry (since GCP seems to be moving in that direction) and it would be good to know if there are still plans for metrics using the Ruby gem. Or if there are any known workarounds or other ruby libraries for reporting metrics to OpenTelemetry.

**Reply by kaylareopelle** [reply]
Hi @jensljungblad-kisi! Thanks for reaching out.

We're making good progress with a Metrics implementation for Ruby. It's not stable yet, but we have experimental implementations of the metrics API, SDK and OTLP exporter available on Ruby gems. Experimental here means:
* The gems aren't fully spec compliant
* We haven't had an OpenTelemetry Technical Committee member evaluate the implementation 

The best way to help us move toward stability is to test out our current gems and give us feedback on how they work in your environment.

The [README for the Metrics SDK](https://github.com/open-telemetry/opentelemetry-ruby/blob/main/metrics_sdk/README.md) lists most of the currently available features and how you can use them in your application.

You may also be interested in the [OTLP exporter for Metrics too.](https://github.com/open-telemetry/opentelemetry-ruby/blob/main/exporter/otlp-metrics/README.md)

The [spec compliance matrix](https://github.com/open-telemetry/opentelemetry-specification/blob/main/spec-compliance-matrix.md#metrics) also has a good list of the features we support.

Our [dice roller](https://github.com/open-telemetry/opentelemetry-ruby/tree/main/examples/dice_roller) example demonstrates a strategy to install and use Metrics in a Sinatra application.

We have not started to incorporate Metrics into our instrumentation, but hope to do that soon.

**Reply by jensljungblad-kisi** [reply]
Thank you. Do you know anything about a timeline reg. getting metrics past the `development` stage?
